File: Python/Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @return a ListNode
    def addTwoNumbers(self, l1, l2):
        ret = ListNode(0)
        newlist = ret
        summary = 0
        while l1 or l2:
            summary //= 10
            if l1:
                summary += l1.val
                l1 = l1.next
            if l2:
                summary += l2.val
                l2 = l2.next
            
            newlist.next = ListNode(summary % 10)
            newlist = newlist.next
        if summary//10 == 1:
            newlist.next = ListNode(1)
        return ret.next
                
        
        

File: Java/src/Add Two Numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode ret = new ListNode(0);
        ListNode newList = ret;
        
        int summary = 0;
        while (l1 != null || l2 !=null){
            summary /= 10;
            if (l1 != null){
                summary += l1.val;
                l1 = l1.next;
            }
            if (l2 !=null){
                summary += l2.val;
                l2 = l2.next;
            }
            newList.next = new ListNode(summary % 10);
            newList = newList.next;
            if (summary/10 == 1){
                newList.next = new ListNode(1);
            }
        }
      return ret.next;  
    }
}